crates/holochain/src/conductor/conductor.rs

```diff
@@ -1400,15 +1402,31 @@ mod app_impls {
                 self.clone().register_dna(dna).await?;
             }
 
-            crate::conductor::conductor::genesis_cells(self.clone(), cells_to_create).await?;
+            let cell_ids: Vec<_> = cells_to_create
+                .iter()
+                .map(|(cell_id, _)| cell_id.clone())
+                .collect();
 
-            let roles = ops.role_assignments;
-            let app = InstalledAppCommon::new(installed_app_id, agent_key, roles, manifest)?;
+            let genesis_result =
+                crate::conductor::conductor::genesis_cells(self.clone(), cells_to_create).await;
 
-            // Update the db
-            let stopped_app = self.add_disabled_app_to_db(app).await?;
+            if genesis_result.is_ok() || ignore_genesis_failure {
+                let roles = ops.role_assignments;
+                let app = InstalledAppCommon::new(installed_app_id, agent_key, roles, manifest)?;
 
-            Ok(stopped_app)
+                // Update the db
+                let stopped_app = self.add_disabled_app_to_db(app).await?;
+
+                // Return the result, which be may an error if no_rollback was specified
+                genesis_result.map(|()| stopped_app)
+            } else if let Err(err) = genesis_result {
+                // Rollback created cells on error
+                dbg!("removed cells");
+                self.remove_cells(&cell_ids).await;
+                Err(err)
+            } else {
+                unreachable!()
+            }
```

Comment:
This is an interesting bit to review

Reply:
LGTM
I suggest to take the dbg statement out or replace it with a trace log